DOC: update Contributor Quickstart guide to refer to the Building From Source Guide

chitvs opened the pull request:
<!-- 
Thanks for contributing a pull request! Please ensure that
your PR satisfies the checklist before submitting:
https://scipy.github.io/devdocs/dev/contributor/development_workflow.html#checklist-before-submitting-a-pr

Also, please name and describe your PR as you would write a
commit message:
https://scipy.github.io/devdocs/dev/contributor/development_workflow.html#writing-the-commit-message.
However, please only include an issue number in the description, not the title,
and please ensure that any code names containing underscores are enclosed in backticks.

Depending on your changes, you can skip CI operations and save time and energy: 
https://scipy.github.io/devdocs/dev/contributor/continuous_integration.html#skipping

Note that we are a team of volunteers; we appreciate your
patience during the review process.

Again, thanks for contributing!
-->

#### Reference issue
Closes #21421.

#### What does this implement/fix?
Updates the `dev-quickstart` guide to refer directly to the "Building From Source for SciPy Development" guide instead of duplicating the instructions.

#### Additional information
This change helps to reduce maintenance burden and ensures that the information in the `dev-quickstart` guide is consistent with the more up-to-date instructions.

chitvs committed:
DOC: update Contributor Quickstart guide to refer to the Building From Source guide

The Contributor Quickstart guide has been updated to point to the more recent and comprehensive Building From Source guide. This change reduces duplication of information and ensures that contributors have access to the most up-to-date instructions.

See #21421.

tylerjereddy commented on doc/source/dev/dev_quickstart.rst:
```diff
@@ -16,8 +16,9 @@ steps to start contributing:
-   :ref:`building-from-source`.
+   dependencies to complete this step depending on your system. For detailed
+   instructions on building SciPy from source, please refer to the 
+   `Building From Source for SciPy Development <https://scipy.github.io/devdocs/building/index.html#building-from-source-for-scipy-development>`_ guide.
```
Wouldn't this only reduce maintenance burden if we actually delete the older/duplicated instructions about `Conda env` and `Virtual env` below? If I understand Lucas correctly, the point is that we longer want to maintain the extra/older instructions below.

chitvs replied:
Alright, got it. So, you want me to remove the duplicated Conda and Virtual environment instructions entirely and just direct people to the Building SciPy From Source guide? Let me know if that's the plan, and I'll make those changes.

lucascolley replied:
Yeah, what I had in mind was replacing the section that starts with "execute the
following commands" with a link to the building from source guide. I suppose it is already linked on line 20 though, so maybe there is a better way to restructure the page! The main thing is to remove the conda/venv instructions from the page.

chitvs replied:
Ok, there we go.

chitvs committed:
DOC: update Contributor Quickstart guide to refer to the Building From Source guide

The Contributor Quickstart guide has been updated to point to the more recent and comprehensive Building From Source guide. This change removes duplicated Conda and Virtual Environment setup instructions and ensures that contributors have access to the most up-to-date information.

See #21421.

chitvs committed:
Merge branch 'scipy:main' into update-quickstart-guide

lucascolley committed:
update wording

[docs only]

lucascolley reviewed (approved):
thanks @chitvs, I pushed some minor changes and I'm happy with this now!

@tylerjereddy please squash merge if this reads okay and the doc build looks good!

chitvs commented:
Hi @tylerjereddy and @lucascolley, is there anything else needed to finalize this PR? Thanks!

Best,
Alessandro (chitvs)

lucascolley commented:
thanks for your first contribution to SciPy @chitvs !
